### ISSUE TITLE
[C#] Implement new Concept Exercise: classes

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1014]
This issue describes how to implement the `classes` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Classes in C#.

## Learning objectives

- Know what classes are.
- Know what encapsulation is.
- Know what fields are.
- Know how to create an object.
- Know how to update state through methods.
- Know about the `void` type.

## Out of scope

- Reference equality.
- Reference parameter passing.
- Constructors.
- Interfaces.
- Inheritance.
- Properties.
- Indexers.
- Structs.
- Destructors.
- Method overloading.
- Pattern matching.

## Concepts

This Concepts Exercise's Concepts are:

- `classes`: know what classes are; know what encapsulation is; know what fields are; know how to define a parameterless constructor; know how to create an object; know how to update state through methods; know about the `void` type.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `basics`: know how to define a basic class with basic methods.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Classes][classes]: tutorial introducing the various aspects of classes.
- [Constructors][constructors]: shows how to define constructors.
- [Creating objects][creating-objects]: shows how to create objects.
- [Fields][fields]: description of class fields.
- [void][void]: describes the `void` type.

### After

- [Classes][classes]: tutorial introducing the various aspects of classes.
- [Constructors][constructors]: shows how to define constructors.
- [Creating objects][creating-objects]: shows how to create objects.
- [Fields][fields]: description of class fields.
- [void][void]: describes the `void` type.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/classes.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[classes]: https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/classes
[visibility-public]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/public
[visibility-private]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private
[creating-objects]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/classes#creating-objects
[fields]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields
[constructors]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors
[void]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/void

**Comment by ErikSchierboom** [comment]
I'm working on this.

**Comment by ErikSchierboom** [comment]
Closed by #1374
